WrightTools/_group.py

```diff
@@ -178,21 +179,113 @@ def parent(self):
-            self.__class__.instances.pop(self.fullpath)
-            self.file.flush()
-            self.file.close()
-            if hasattr(self, '_tmpfile'):
-                os.close(self._tmpfile[0])
-                os.remove(self._tmpfile[1])
+            self.__class__.instances.pop(self.fullpath, None)
+            try:
+                self.file.flush()
+                self.file.close()
+            except SystemError:
+                pass
+            finally:
+                if hasattr(self, '_tmpfile'):
+                    os.close(self._tmpfile[0])
+                    os.remove(self._tmpfile[1])
+
+    def copy(self, parent=None, name=None, verbose=True):
+        """Create a copy under parent.
+
+        All children are copied as well.
+
+        Parameters
+        ----------
+        parent : WrightTools Collection (optional)
+            Parent to copy within. If None, copy is created in root of new
+            tempfile. Default is None.
+        name : string (optional)
+            Name of new copy at destination. If None, the current natural
+            name is used. Default is None.
+        verbose : boolean (optional)
+            Toggle talkback. Default is True.
+
+        Returns
+        -------
+        Group
+            Created copy.
+        """
+        if name is None:
+            name = self.natural_name
+        if parent is None:
+            new = Group()
+            # attrs
+            for k, v in self.attrs.items():
+                new.attrs[k] = v
+            new.natural_name = name
+            # children
+            for k in self.keys():
+                self[k].copy(new)
+            new.flush()
+            p = new.filepath
+            from ._open import open as wt_open
+            new = wt_open(p)
+        else:
+            # copy
+            self.file.copy(self.name, parent, name=name)
+            if 'item_names' in parent.attrs.keys():
+                new = parent.item_names + (name,)
+                parent.attrs['item_names'] = np.array(new, dtype='S')
+            parent._update_natural_namespace()
+            new = parent[name]
+        # finish
+        if verbose:
+            print('{0} copied to {1}'.format(self.fullpath, new.fullpath))
+        return new
 
     def flush(self):
         """Ensure contents are written to file."""
         self.file.flush()
+
+    def save(self, filepath=None, verbose=True):
+        """Save as root of a new file.
+
+        Parameters
+        ----------
+        filepath : string (optional)
+            Filepath to write. If None, file is created using natural_name.
+        verbose : boolean (optional)
+            Toggle talkback. Default is True
+
+        Returns
+        -------
+        str
+            Written filepath.
+        """
+        # parse filepath
+        if filepath is None:
+            filepath = os.path.join(os.getcwd(), self.natural_name + '.wt5')
+        elif len(os.path.basename(filepath).split('.')) == 1:
```

Comment:
Do we wish to have a more robust check here?
I would tend to use other characters myself, but I can imagine a scenario where someone wants a dot in their file name, but still wants the extension added. e.g. saving a sequence of related files (single collection may be preferred within the WT ecosystem, but if interacting with other things...) as `name.000.wt5` `name.001.wt5` etc.

Perhaps we check that the last thing after the `.` is not in a list of known hdf5 subtype file extensions: (e.g. h5 hdf5 wt5 netcdf nc) etc.

I think it may be easiest to not try and cover them all, but cover the bare hdf5 extensions and wt5 to check that they are already there, otherwise add the wt5 extension.

Obviously users have the ability to change the extension externally, but still, it can be confusing.